src/analyzing/Constant_propagation.ml

```diff
@@ -103,29 +103,37 @@ let rec lvars_in_lhs expr =
   | Container ((Tuple | Array), (_, es, _)) -> List.concat_map lvars_in_lhs es
   | __else__ -> []
 
+class ['self] no_cycles_in_sym_prop_visitor =
```

Comment:
is this covered in a test? assume so :)

Reply:
We ok with this? We did not change the logic here, just refactored a bit the code.